### DISCUSSION TITLE
Add IndexMove() to List

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/runtime#110562] [Ideas]
The effect is similar to
```csharp
var temp = list[index];
list.RemoveAt(index);
list.Insert(newIndex, temp);
```
This operation will cause all subsequent projects after the affected project to move meaningless twice.
I hope he has his own API to achieve this more efficiently.

**Comment by huoyaoyuan** [comment]
For efficiency, you can use `CollectionsMarshal.AsSpan` to manipulate the backing storage yourself. The operation doesn't seem to be common.

**Comment by redflitzi** [comment]
Hm, a first step might be to invent for `List <T>`:

```
public T Detach(T item)
public T DetachAt(int index)

```
where both methods return the detached list member as a shallow copy, or maybe as a reference (when reference type), instead of just deleting it like Remove()/RemoveAt() do.

Then, in your case, the operation will shrink to

```
var temp = list.DetachAt(index);
list.Insert(newindex, temp);

```
This might make the intention more clear.

**Reply by huoyaoyuan** [reply]
It won't work so. A linear list is not a linked list and elements can't be "detached" separately.

**Comment by StellarWolfEntertainment** [comment]
I know it's not exactly what you are asking but you could always right it as an extension method [at least until C# adds this if they do]
```csharp
public static void IndexMove<T>(this List<T> list, int index, int newIndex) {
    T element = list[index];
    list.RemoveAt(index);
    list.Insert(newIndex, element);
}
```
but I do agree that this is a fairly major concept that is missing from lists in C#
